// @u4/opencv-build/esm/src/StaticTools.js
import * as dntShim from "../_dnt.shims.js";
import fs from "node:fs";
import os from "node:os";
import path from "node:path";
import { getDirname, getEnv, setEnv } from "./env.js";
import * as detector from "./helper/detect.js";
import Log from "./Log.js";
import { highlight } from "./utils.js";
class StaticTools {
    constructor() {
        Object.defineProperty(this, "readEnvsFromPackageJsonLog", {
            enumerable: true,
            configurable: true,
            writable: true,
            value: 0
        });
    }
    /**
     * Find the proper root dir, this directory will contains all openCV source code and a subfolder per build
     * @param opts
     * @returns
     */
    getBuildDir(opts = {}) {
        let buildRoot = opts.buildRoot || getEnv("OPENCV_BUILD_ROOT") ||
            path.join(getDirname(), "..");
        if (buildRoot[0] === "~") {
            buildRoot = path.join(os.homedir(), buildRoot.slice(1));
        }
        return buildRoot;
    }
    getPackageJson() {
        // return path.resolve(process.cwd(), "package.json");
        return dntShim.Deno.realPathSync(dntShim.Deno.cwd() + "/package.json");
    }
    /**
     * autodetect path using common values.
     * @return number of updated env variable from 0 to 3
     */
    autoLocatePrebuild() {
        let changes = 0;
        const summery = [];
        if (!getEnv("OPENCV_BIN_DIR")) {
            const candidate = detector.detectBinDir();
            if (candidate) {
                setEnv("OPENCV_BIN_DIR", candidate);
                changes++;
            }
        }
        if (!getEnv("OPENCV_LIB_DIR")) {
            const candidate = detector.detectLibDir();
            if (candidate) {
                setEnv("OPENCV_LIB_DIR", candidate);
                changes++;
            }
        }
        if (!getEnv("OPENCV_INCLUDE_DIR")) {
            const candidate = detector.detectIncludeDir();
            if (candidate) {
                setEnv("OPENCV_INCLUDE_DIR", candidate);
                changes++;
            }
        }
        return { changes, summery };
    }
    /**
     * list existing build in the diven directory
     * @param rootDir build directory
     * @returns builds list
     */
    listBuild(rootDir) {
        let workFolderContent = [];
        try {
            workFolderContent = fs.readdirSync(rootDir);
        }
        catch (err) {
            throw new Error("Failed to list directory: " + rootDir +
                " Check environement variable OPENCV_BUILD_ROOT, " + err);
        }
        const versions = workFolderContent
            .filter((n) => n.startsWith("opencv-"))
            .map((dir) => {
            const autobuild = path.join(rootDir, dir, "auto-build.json");
            try {
                const stats = fs.statSync(autobuild);
                const hash = dir.replace(/^opencv-.+-/, "-");
                const buildInfo = this.readAutoBuildFile(autobuild, true);
                return { autobuild, dir, hash, buildInfo, date: stats.mtime };
            }
            catch (_err) {
                return {
                    autobuild,
                    dir,
                    hash: "",
                    buildInfo: null,
                    date: 0,
                };
            }
        })
            .filter((n) => n.buildInfo);
        return versions;
    }
    /**
     * Read a parse an existing autoBuildFile
     * @param autoBuildFile file path
     * @returns
     */
    readAutoBuildFile(autoBuildFile, quiet) {
        try {
            const fileExists = fs.existsSync(autoBuildFile);
            if (fileExists) {
                const autoBuildFileData = JSON.parse(fs.readFileSync(autoBuildFile).toString());
                if (!autoBuildFileData.opencvVersion ||
                    !("autoBuildFlags" in autoBuildFileData) ||
                    !Array.isArray(autoBuildFileData.modules)) {
                    // if (quiet) return undefined;
                    throw new Error(`auto-build.json has invalid contents, please delete the file: ${autoBuildFile}`);
                }
                return autoBuildFileData;
            }
            if (!quiet) {
                Log.log("info", "readAutoBuildFile", "file does not exists: %s", autoBuildFile);
            }
        }
        catch (err) {
            //if (!quiet)
            if (err instanceof Error) {
                Log.log("error", "readAutoBuildFile", "failed to read auto-build.json from: %s, with error: %s", autoBuildFile, err.toString());
            }
            else {
                Log.log("error", "readAutoBuildFile", "failed to read auto-build.json from: %s, with error: %s", autoBuildFile, JSON.stringify(err));
            }
        }
        return undefined;
    }
    /**
     * extract opencv4nodejs section from package.json if available
     */
    parsePackageJson() {
        const absPath = this.getPackageJson();
        if (!fs.existsSync(absPath)) {
            return null;
        }
        const data = JSON.parse(fs.readFileSync(absPath).toString());
        return { file: absPath, data };
    }
    /**
     * get opencv4nodejs section from package.json if available
     * @returns opencv4nodejs customs
     */
    readEnvsFromPackageJson() {
        const rootPackageJSON = this.parsePackageJson();
        if (!rootPackageJSON) {
            return null;
        }
        if (!rootPackageJSON.data) {
            if (!this.readEnvsFromPackageJsonLog++) {
                Log.log("info", "config", `looking for opencv4nodejs option from ${highlight("%s")}`, rootPackageJSON.file);
            }
            return {};
        }
        if (!rootPackageJSON.data.opencv4nodejs) {
            if (!this.readEnvsFromPackageJsonLog++) {
                Log.log("info", "config", `no opencv4nodejs section found in ${highlight("%s")}`, rootPackageJSON.file);
            }
            return {};
        }
        if (!this.readEnvsFromPackageJsonLog++) {
            Log.log("info", "config", `found opencv4nodejs section in ${highlight("%s")}`, rootPackageJSON.file);
        }
        return rootPackageJSON.data.opencv4nodejs;
    }
}
const singleton = new StaticTools();
export default singleton;
