// @denodnt/logger/script/writer.js
"use strict";
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
var _a, _b, _c;
Object.defineProperty(exports, "__esModule", { value: true });
const dntShim = __importStar(require("./_dnt.shims.js"));
const writable_js_1 = __importDefault(require("./writable.js"));
const fs_js_1 = require("./fs.js");
const types_js_1 = __importDefault(require("./types.js"));
/**
 * Writer class
 */
class Writer {
    /**
     * Writer constructor
     * @param param0
     * @returns
     */
    constructor({ maxBytes, maxBackupCount }) {
        Object.defineProperty(this, "maxBytes", {
            enumerable: true,
            configurable: true,
            writable: true,
            value: void 0
        });
        Object.defineProperty(this, "maxBackupCount", {
            enumerable: true,
            configurable: true,
            writable: true,
            value: void 0
        });
        Object.defineProperty(this, "pathWriterMap", {
            enumerable: true,
            configurable: true,
            writable: true,
            value: new Map()
        });
        Object.defineProperty(this, _a, {
            enumerable: true,
            configurable: true,
            writable: true,
            value: ""
        });
        Object.defineProperty(this, _b, {
            enumerable: true,
            configurable: true,
            writable: true,
            value: ""
        });
        Object.defineProperty(this, _c, {
            enumerable: true,
            configurable: true,
            writable: true,
            value: ""
        });
        if (maxBytes !== undefined && maxBytes <= 0) {
            throw new Error("maxBytes cannot be less than 1");
        }
        this.maxBytes = maxBytes;
        if (maxBackupCount === undefined)
            return;
        if (!maxBytes) {
            throw new Error("maxBackupCount must work with maxBytes");
        }
        if (maxBackupCount <= 0) {
            throw new Error("maxBackupCount cannot be less than 1");
        }
        this.maxBackupCount = maxBackupCount;
    }
    async newWriter(path) {
        const writer = new writable_js_1.default(path);
        await writer.setup();
        this.pathWriterMap.set(path, writer);
        return writer;
    }
    /**
     * Write message to file
     * @param param0
     * @returns
     */
    async write({ path, msg, type }) {
        const msgByteLength = msg.byteLength;
        if (this.pathWriterMap.has(path)) {
            const writer = this.pathWriterMap.get(path);
            const currentSize = writer.currentSize;
            const size = currentSize + msgByteLength;
            if (this.maxBytes && size > this.maxBytes) {
                writer.close();
                this.rotateLogFiles(path);
                const _writer = await this.newWriter(path);
                await _writer.write(msg);
            }
            else {
                await writer.write(msg);
            }
            return;
        }
        const typePath = this[type];
        if (typePath) {
            const pathWriter = this.pathWriterMap.get(typePath);
            if (pathWriter && pathWriter.close)
                pathWriter.close();
            this.pathWriterMap.delete(typePath);
        }
        this[type] = path;
        const writer = await this.newWriter(path);
        await writer.write(msg);
    }
    /**
     * Rotate log files
     * @param path
     */
    async rotateLogFiles(path) {
        if (this.maxBackupCount) {
            for (let i = this.maxBackupCount - 1; i >= 0; i--) {
                const source = path + (i === 0 ? "" : "." + i);
                const dest = path + "." + (i + 1);
                const exist = await (0, fs_js_1.exists)(source);
                if (exist) {
                    await dntShim.Deno.rename(source, dest);
                }
            }
        }
        else {
            const dest = path + "." + Date.now();
            await dntShim.Deno.rename(path, dest);
        }
    }
}
_a = types_js_1.default.INFO, _b = types_js_1.default.WARN, _c = types_js_1.default.ERROR;
exports.default = Writer;
